arbitrator/prover/src/machine.rs

```diff
@@ -2567,10 +2641,14 @@ impl Machine {
             |v| v.serialize_for_proof(),
         ));
 
-        out!(prove_window(
-            self.get_frame_stack(),
+        out!(prove_multistack(
+            self.get_frame_stacks(),
             hash_stack_frame_stack,
-            StackFrame::serialize_for_proof,
+            |stack| prove_window(
+                stack, 
+                hash_stack_frame_stack, 
+                StackFrame::serialize_for_proof
+            ),
         ));
 
         out!(self.prove_guards());
```

Comment:
another addition necessary: in the end of this function there are additions to proof by opcode.
In case the opcode is PopCoThread - 4 hashes need to be added to the proof:
the hash of new last stack, followed by the new remaining hashes - first for value multistack, then for frame multistack. Each of those might again be -1 if empty.

Reply:
Done